lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -361,16 +361,8 @@ public function getListTableIndexesSQL($table, $currentDatabase = null)
      */
     private function getTableWhereClause($table, $classAlias = 'c', $namespaceAlias = 'n')
     {
-        $whereClause = $namespaceAlias.".nspname NOT IN ('pg_catalog', 'information_schema', 'pg_toast') AND ";
-        if (strpos($table, ".") !== false) {
-            list($schema, $table) = explode(".", $table);
-            $schema = "'" . $schema . "'";
-        } else {
-            $schema = "ANY(string_to_array((select replace(replace(setting,'\"\$user\"',user),' ','') from pg_catalog.pg_settings where name = 'search_path'),','))";
-        }
-
-        $table = new Identifier($table);
-        $whereClause .= "$classAlias.relname = '" . $table->getName() . "' AND $namespaceAlias.nspname = $schema";
+        $tableNameEscaped = str_replace("'", "''", $table);
+        $whereClause = $classAlias.".oid='".$tableNameEscaped."'::regclass::oid";
```

Comment:
Neat! So much simpler then :) The only thing we'll have to take care of then is handling incoming quoted identifiers:

```php
private function getTableWhereClause($table, $classAlias = 'c', $namespaceAlias = 'n')
{
    $table = new Identifier($table);
    $table = $this->quoteStringLiteral($table->getName());
    
    return sprintf('%s.oid = %s::regclass::oid', $classAlias, $table);
}
```

Something like that. Tests should also cover some of these cases. Quoted/unquoted, qualified/unqualified, special characters/regular characters